pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BrokerRequestHandler.java

```diff
@@ -581,15 +580,15 @@ private BrokerResponse processOptimizedBrokerRequests(@Nonnull BrokerRequest ori
    * @return server response map.
    */
   @Nullable
-  private Map<ServerInstance, ByteBuf> gatherServerResponses(@Nonnull CompositeFuture<ByteBuf> compositeFuture,
+  private Map<ServerInstance, byte[]> gatherServerResponses(@Nonnull CompositeFuture<byte[]> compositeFuture,
       @Nonnull ScatterGatherStats scatterGatherStats, boolean isOfflineTable, @Nonnull String tableNameWithType,
       @Nonnull List<ProcessingException> processingExceptions) {
     try {
-      Map<ServerInstance, ByteBuf> serverResponseMap = compositeFuture.get();
-      Iterator<Entry<ServerInstance, ByteBuf>> iterator = serverResponseMap.entrySet().iterator();
+      Map<ServerInstance, byte[]> serverResponseMap = compositeFuture.get();
+      Iterator<Entry<ServerInstance, byte[]>> iterator = serverResponseMap.entrySet().iterator();
       while (iterator.hasNext()) {
-        Entry<ServerInstance, ByteBuf> entry = iterator.next();
-        if (entry.getValue().readableBytes() == 0) {
+        Entry<ServerInstance, byte[]> entry = iterator.next();
+        if (entry.getValue().length == 0) {
```

Comment:
not sure if you want entry.getValue() == null test as well here?

Reply:
It can never be null here. If it is null, then just let it throw NPE